Source/AGS.API/Graphics/Animations/IAnimationState.cs

```diff
@@ -38,6 +38,12 @@ public interface IAnimationState : INotifyPropertyChanged
         /// <value><c>true</c> if is paused; otherwise, <c>false</c>.</value>
 		bool IsPaused { get; set; }
 
+        /// <summary>
+        /// Allows to subscribe to the animation frame change.
+        /// </summary>
+        // TODO: Should this be a blocking event?
+        IEvent OnNextFrame { get; }
```

Comment:
You don't need to add this event: IAnimationState implements INotifyPropertyChanged, so you can use the PropertyChanged event instead, and check for the CurrentFrame property change:

```csharp

animationState.PropertyChanged += onAnimationStatePropertyChanged;

private void onAnimationStatePropertyChanged(object sender, PropertyChangedEventArgs args)
{
   if (args.PropertyName != nameof(IAnimationState.CurrentFrame)) return;
   
   //... change sprite here ...
}

```

Note that you will not find the code that actually fires that event, that's because I'm using https://github.com/Fody/PropertyChanged for injecting the code on compile.